### PR TITLE
James 3544 webadmin upload repository cleanup

### PR DESCRIPTION
**Pull request by vttranlina**
Ref : https://issues.apache.org/jira/browse/JAMES-3544

https://github.com/linagora/james-project/issues/4357

### CODE REVIEW

**Comment by vttranlina** [comment]
Has a problem with `org.apache.james.jmap.cassandra.upload.CassandraUploadRepository#purge` method when delete list bucket.

**Review by chibenwa — COMMENTED** [review]
Looks great to me!

**Comment by chibenwa** [comment]
`The file /home/jenkins/jenkins-agent/workspace/james_ApacheJames_PR-590/server/protocols/webadmin/webadmin-jmap/pom.xml is not sorted`

Good luck, I hate that crap!

**Comment by chibenwa** [comment]
> Has a problem with org.apache.james.jmap.cassandra.upload.CassandraUploadRepository#purge method when delete list bucket. 

Can you be more precise?

**Comment by vttranlina** [comment]
> > Has a problem with org.apache.james.jmap.cassandra.upload.CassandraUploadRepository#purge method when delete list bucket.
> 
> Can you be more precise?

You can run `org.apache.james.webadmin.data.jmap.JmapUploadRoutesTest#cleanUploadTaskShouldSuccessWhenMixCase`
It removes only one expired bucket. (My expect: It have to remove two expired buckets)

File: server/protocols/webadmin/webadmin-jmap/src/main/java/org/apache/james/webadmin/data/jmap/UploadRepositoryCleanupTask.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    enum CleanupScope {
+        EXPIRED;
+
+        static class CleanupScopeInvalidException extends IllegalArgumentException {
+        }
+
+        public static Optional<CleanupScope> from(String name) {
+            Preconditions.checkNotNull(name);
+            return Arrays.stream(CleanupScope.values())
+                .filter(value -> name.equalsIgnoreCase(EXPIRED.name()))
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-jmap/src/main/java/org/apache/james/webadmin/data/jmap/UploadRepositoryCleanupTask.java:54]:
```suggestion
                .filter(value -> name.equalsIgnoreCase(value))
```
Or what is the point or streaming values if you check against the same one each time in your stream? Even if there is only one value :)

**Comment by Arsnael** [comment]
> You can run `org.apache.james.webadmin.data.jmap.JmapUploadRoutesTest#cleanUploadTaskShouldSuccessWhenMixCase`
> It removes only one expired bucket. (My expect: It have to remove two expired buckets)

Not  sure what is wrong but could possibly be your mocks on the Clock (mocks are evil^^). If you apply @chibenwa suggestion to use `UpdatableTickingClock`, you have a better chance of having your test working IMO. Try this first, if still not working then we will need to dig more :)

**Comment by chibenwa** [comment]
Write more tests for the `BucketNameGenerator` too maybe?

**Comment by vttranlina** [comment]
> > You can run `org.apache.james.webadmin.data.jmap.JmapUploadRoutesTest#cleanUploadTaskShouldSuccessWhenMixCase`
> > It removes only one expired bucket. (My expect: It have to remove two expired buckets)
> 
> Not sure what is wrong but could possibly be your mocks on the Clock (mocks are evil^^). If you apply @chibenwa suggestion to use `UpdatableTickingClock`, you have a better chance of having your test working IMO. Try this first, if still not working then we will need to dig more :)

I tried `UpdatableTickingClock` on local, It still not works

**Comment by Arsnael** [comment]
Can you push what you tried @vttranlina ?

**Comment by vttranlina** [comment]
> Write more tests for the `BucketNameGenerator` too maybe?

I'm not sure, I have just check it, it is already, `BucketNameGeneratorTest`, 
I tried more tests, but IMO it's duplicated with the old test, maybe unnecessary!

**Comment by chibenwa** [comment]
https://github.com/apache/james-project/pull/590/commits/b2d9f09463d911e87dbed4cd6ee8ae003ff57bb4 especially is a best practice we should not have had been missing!

**Comment by chibenwa** [comment]
```
Unable to create injector, see the following errors:

1) [Guice/MissingImplementation]: No implementation for Session was bound.

Requested by:
1  : UploadDAO.<init>(UploadDAO.java:134)
```

Guice bindings needs to be cleaned up. We don't want this endpoint for memory.

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Can you squash the fixups @vttranlina ?

**Review by chibenwa — APPROVED** [review]

**Comment by vttranlina** [comment]
Rebase master & squash fixup

**Pull request merged**